src/osd/ReplicatedPG.cc

```diff
@@ -1673,17 +1673,17 @@ void ReplicatedPG::do_op(OpRequestRef& op)
   // discard due to cluster full transition?  (we discard any op that
   // originates before the cluster or pool is marked full; the client
   // will resend after the full flag is removed or if they expect the
-  // op to succeed despite being full).  The except is FULL_FORCE ops,
-  // which there is no reason to discard because they bypass all full
+  // op to succeed despite being full).  The except is FULL_FORCE|FULL_TRY
+  // ops, which there is no reason to discard because they bypass all full
   // checks anyway.
-  // FIXME: we exclude mds writes for now.
-  if (!(m->get_source().is_mds() || m->has_flag(CEPH_OSD_FLAG_FULL_FORCE)) &&
+  if (!m->has_flag(CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_TRY) &&
       info.history.last_epoch_marked_full > m->get_map_epoch()) {
     dout(10) << __func__ << " discarding op sent before full " << m << " "
 	     << *m << dendl;
     return;
   }
-  if (!m->get_source().is_mds() && osd->check_failsafe_full()) {
+  if (!m->has_flag(CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_FORCE) &&
```

Comment:
This needs to be CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_TRY.  I wish this code looked like this:

<pre>
if (!m->has_flag(CEPH_OSD_FLAG_FULL_FORCE|CEPH_OSD_FLAG_FULL_TRY) {
    if (info.history.last_epoch_marked_full > m->get_map_epoch()) {
        dout(...
        return;
    }
    if (osd->check_failsafe_full()) {
        dout(...
        return;
    }
}
</pre>